### DISCUSSION TITLE
OpenVPN and IPsec (IKEv2)

### DISCUSSION THREAD

**Discussion by bhutkovskyy** [strongswan/strongswan#516] [Q&A]
Hi,

I am trying to set up additional strongswan routing for my OpenVPN clients and would like to know from experts if I am on a right track.

Currently, my VPN clients can connect to cloud OpenVPN server and route to the internal subnet (10.1.10.0/24) just fine. However, we have another VPN server (on-prem) running strongswan and my approach was to add an ipsec connection from OpenVPN access server to this strongswan server.

I was able to set up this connection on OpenVPN server itself and now it can reach another private on-prem subnet (10.0.0.0/23). This only works from the openvpn server itself where the ipsec connection is established. Now, I would like my OpenVPN clients to also be able to reach hosts on this other private subnet (10.0.0.0/23). Is it possible? And if so, how do I set up routing for clients so they can reach this on-prem subnet, which is using ipsec VPN connection. The OpenVPN community directed my here since this seems to be more of an issue on the strongswan/ipsec side and not related to OpenVPN itself.

Server info (where OpenVPN AS runs and where I have strongswan client installed):

Ubuntu 18.04
Linux strongSwan U5.6.2/K5.4.0-1051-azure

I am using NAT setup in OpenVPN. Here's what I get when I try to ping a host in (10.0.0.0/23) subnet:

```
92 bytes from 172.27.232.1: Destination Host Unreachable
Vr HL TOS  Len   ID Flg  off TTL Pro  cks      Src      Dst
 4  5  00 5400 6844   0 0000  3f  01 7415 172.27.232.2  10.0.1.50
```

Here's what I get with tcpdump:
```
00:21:46.543113 IP 172.27.232.2 > 10.0.1.50: ICMP echo request, id 7241, seq 563, length 64
00:21:49.597564 IP openvpn-01 > 172.27.232.2: ICMP host 10.0.1.50 unreachable, length 92
```

and this is what my ipsec.conf looks like:
```
conn vpn-connection-ikev2
    auto=start
    type=tunnel
    right=x.x.x.x (IP of remote VPN server running strongswan)
    rightid=myremotestrongswanserver.com
    rightsubnet=10.0.0.1/23 (subnet on remote vpn server side)
    rightauth=pubkey
    leftsourceip=%config,172.27.224.0/20 (IP CIDR of OpeVPN clients)
    leftid=somename
    leftauth=pubkey
    leftcert=vpnclient.cer
    leftsubnet=0.0.0.0/0
    ike=aes256gcm16-prfsha512-ecp384!
    esp=aes256gcm16-ecp384!
    dpdaction=restart
    closeaction=restart
```

Thanks so much in advance!

**Comment by tobiasbrunner** [comment]
Please read #196 (and possibly the linked documents e.g. on forwarding traffic) and update your question. But I can already tell you that your setting for `leftsourceip` is not correct (the subnet should probably go to `leftsubnet` but it really depends on what the config on the other end looks like and what your intentions are e.g. whether you want to NAT traffic or just route it).
